### PR TITLE
Support v0 burn event with base58 encoded `to`

### PR DESCRIPTION
**Pull request by tok-kkk**

### CODE REVIEW

**Review by TheAncientGoat — COMMENTED** [review]
One minor issue, otherwise looks good. Will test burning on my end

File: watcher/watcher.go [hunk 1]
```diff
@@ -35,10 +41,11 @@ type Watcher struct {
 }
 
 // NewWatcher returns a new Watcher.
-func NewWatcher(logger logrus.FieldLogger, selector tx.Selector, bindings txengine.Bindings, ethClient *ethclient.Client, ethBindings *ethereumbindings.MintGatewayLogicV1, resolver jsonrpc.Resolver, cache redis.Cmdable, distPubKey *id.PubKey, pollInterval time.Duration) Watcher {
+func NewWatcher(logger logrus.FieldLogger, network multichain.Network, selector tx.Selector, bindings txengine.Bindings, ethClient *ethclient.Client, ethBindings *ethereumbindings.MintGatewayLogicV1, resolver jsonrpc.Resolver, cache redis.Cmdable, distPubKey *id.PubKey, pollInterval time.Duration) Watcher {
 	gpubkey := (*btcec.PublicKey)(distPubKey).SerializeCompressed()
 	return Watcher{
 		logger:       logger,
+		network:  	  network,
```
[the remark this answers is not in the document]
Comment (tok-kkk) [reply watcher/watcher.go]: Thanks. It might be something to do with my IDE as it's showing fine from it.

[the remark this answers is not in the document]
Comment (TheAncientGoat) [reply watcher/watcher.go]: Possibly different tab widths. Gofmt should ensure its consistent though. (also we should standardize .editorconfigs for all projects)

**Review by TheAncientGoat — APPROVED** [review]
Looks good and have tested - burn + mint of zec works, still waiting for bch deposit to confirm

File: watcher/watcher.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (tok-kkk) [reply watcher/watcher.go]: (I'm not using vscode 😅 )

**Comment by TheAncientGoat** [comment]
.editorconfig is pretty widely supported. Think there's even a nano plugin
for it

**Pull request merged**